doc/architecture/decisions/0008-data-manipulation-for-views-should-happen-in-util-files.md

```diff
@@ -0,0 +1,158 @@
+# 7. Data manipulation for views should happen in util files
+
+Date: 2023-02-14
+
+## Status
+
+Proposed
+
+## Context
+
+- Nunjucks is limited as to how much manipulation it allows and therefore some kinds of neccessary manipulation can only happen in Node.
+- Further, manipulation in Nunjucks is only testable with integration tests (with our current test setup) which are slow to run.
+- There isn't a clear precedent in HMPPS as to where such manipulation should happen.
+
+## Decision
+
+We initially did such manipulation in the services but as the app has evolved we now believe that manipulation of data for views that isn't possible in Nunjucks should happen in individual utils which are imported into the Nunjucks templates. These utils should be imported on a file-by-file (as opposed to function-by-function) basis as global into the nunjucksSetup.ts file.
+
+## Consequences
+
+- Our view code will be cleanly seperated from the application code
+- Our nunjucksSetup file will have to import and export a lot of util files
+- The amount of utils files we have will grow at a faster rate
+- Each util function lives seperate to the data that it requires, meaning that it is simpler to test at the unit test level
+- As the utils are imported as whole files without specifying individual functions there is little boilerplate involved in using the utils in the views.
+
+## Examples
+
```

Comment:
this is great, thanks, I really understand the decision and agree way of working now with an example.